src/App.tsx

```diff
@@ -17,18 +71,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterType={filterType}
+                onFilterType={handleSelelect}
+                onQuery={handleQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todoList.length === 0 ? (
+                hasLoadingError ? 'Error: Data has not been loaded' : <Loader />
```

Comment:
It's bad practice to use ternary operator inside another one.
For the learning it's ok but for the future it's better to think about more pure solution.